File: Node/TypeORM_ORM/src/bll/services/shipperService.ts
import { Shippers } from 'dal/entities/Shippers';
import { ShipperRepository } from '../../dal/repositories/shipperRepository';

export class ShipperService {
  private shipperRepository = new ShipperRepository();
  async getAll() {
    return await this.shipperRepository.getAll();
  }

  async getById(id: number) {
    return await this.shipperRepository.getById(id);
  }

  async create(data: Partial<Shippers>) {
    return await this.shipperRepository.create(data);
  }

  async update(id: number, data: Partial<Shippers>) {
    return await this.shipperRepository.update(id, data);
  }

  async delete(id: number) {
    return await this.shipperRepository.delete(id);
  }
}

File: Go/GORM_ORM/bll/services/shipperService.go
package services

import (
	"GORM_ORM/dal/entities"
	"GORM_ORM/dal/repositories"
)

type ShipperService interface {
	GetAll() ([]entities.Shipper, error)
	GetByID(id int) (*entities.Shipper, error)
	Create(shipper *entities.Shipper) error
	Update(shipper *entities.Shipper) error
	Delete(id int) error
}

type shipperService struct {
	repo repositories.ShipperRepository
}

func NewShipperService(repo repositories.ShipperRepository) ShipperService {
	return &shipperService{repo: repo}
}

func (s *shipperService) GetAll() ([]entities.Shipper, error) {
	return s.repo.GetAll()
}

func (s *shipperService) GetByID(id int) (*entities.Shipper, error) {
	return s.repo.GetByID(id)
}

func (s *shipperService) Create(shipper *entities.Shipper) error {
	return s.repo.Create(shipper)
}

func (s *shipperService) Update(shipper *entities.Shipper) error {
	return s.repo.Update(shipper)
}

func (s *shipperService) Delete(id int) error {
	return s.repo.Delete(id)
}
